pubsub/pulsar/pulsar.go

```diff
@@ -317,3 +352,14 @@ func (p *Pulsar) formatTopic(topic string) string {
 	}
 	return fmt.Sprintf(topicFormat, persist, p.metadata.Tenant, p.metadata.Namespace, topic)
 }
+
+func formatDuration(durationString string) (time.Duration, error) {
```

Comment:
@shubham1172 Is there a better way?

Reply:
We can make it simpler by always expecting a number. 

The metadata can be `batchingMaxPublishDelayMs` to clearly indicate that it should be in milliseconds.

Reply:
@berndverst this comment was still pending.

Reply:
@shubham1172 Need to change to such a design?

Reply:
That would have been cleaner and simpler but Bernd has already approved/merged this PR. If we have enough consensus, we can create another PR on top to change that I guess.

Reply:
@shubham1172  I don't have a better idea. I can create a new rp to modify it

Reply:
Thanks @saber-wang. Let's see what Bernd/@daixiang0 have to say.